python/ray/data/_internal/execution/backpressure_policy.py

```diff
@@ -0,0 +1,100 @@
+import logging
+import os
+from abc import ABC, abstractmethod
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from ray.data._internal.execution.interfaces.physical_operator import (
+        PhysicalOperator,
+    )
+    from ray.data._internal.execution.streaming_executor_state import Topology
+
+logger = logging.getLogger(__name__)
+
+
+# TODO(hchen): Enable ConcurrencyCapBackpressurePolicy by default.
+DEFAULT_BACKPRESSURE_PLOCIES = []
+
+
+def get_backpressure_policies(topology: "Topology"):
+    return [policy(topology) for policy in DEFAULT_BACKPRESSURE_PLOCIES]
+
+
+class BackpressurePolicy(ABC):
+    """Interface for back pressure policies."""
+
+    @abstractmethod
+    def __init__(self, topology: "Topology"):
+        ...
+
+    @abstractmethod
+    def can_run(self, op: "PhysicalOperator") -> bool:
+        """Called when StreamingExecutor is about to select an operator to run.
```

Comment:
Shall we be more specific for which function will be calling it, such as 

```
Called when StreamingExecutor selects an operator to run in `streaming_executor_state.select_operator_to_run()`.
Returns True if the operator can run, False otherwise.
```